aten/src/ATen/native/mps/operations/BinaryKernel.mm

```diff
@@ -264,4 +294,19 @@ static void copysign_mps_kernel(TensorIteratorBase& iter) {
 REGISTER_DISPATCH(fmin_stub, &fmin_mps_kernel);
 REGISTER_DISPATCH(copysign_stub, &copysign_mps_kernel);
 
+Tensor& polar_out_mps(const Tensor& abs, const Tensor& angle, Tensor& output) {
+  auto new_size = at::infer_size(abs.sizes(), angle.sizes());
+  if (!output.sizes().equals(new_size)) {
+    output.resize_(new_size);
+  }
+  uint32_t length = output.numel();
+  if (length == 0) {
+    return output;
+  }
+  auto output_as_real = at::view_as_real(output).index({at::indexing::Ellipsis, 0});
```

Comment:
What is that index() call for. I would suggest using narrow/select/view/reshape in c++ as they are a lot more reliable in their behavior.

Reply:
Replaced with `.select(output.dim(), 0)`